File: src/crazy.ts
import fs from 'fs'
import path from 'path'

function getAllFiles(directoryPath: string): string[] {
  const files: string[] = []
  fs.readdirSync(directoryPath).forEach((file: string) => {
    const filePath = path.join(directoryPath, file)
    if (fs.statSync(filePath).isDirectory()) {
      files.push(...getAllFiles(filePath))
    } else if (
      filePath.endsWith('.ts') &&
      !filePath.endsWith(path.basename(process.argv[1]))
    ) {
      files.push(filePath)
    }
  })
  return files
}

const folderPath = '.'
const allFiles = getAllFiles(folderPath)

let outputString = ''
allFiles.forEach((filePath: string) => {
  const contents = fs.readFileSync(filePath, { encoding: 'utf8' })
  outputString += `\n\n\n\n/* ${filePath} */\n${contents}\n/*\n`
})

fs.writeFileSync('full_code.txt', outputString)

File: crazy.py
import os
import sys

def get_all_py_files(path):
    files = []
    for dirpath, dirnames, filenames in os.walk(path):
        for filename in filenames:
            if filename.endswith('.py') and not filename == os.path.basename(sys.argv[0]):
                files.append(os.path.join(dirpath, filename))
    return files

folder_path = "."
all_files = get_all_py_files(folder_path)

output_string = ""
for file_path in all_files:
    with open(file_path, 'r') as file:
        contents = file.read()
        output_string += f"\n\n\n\n''' {file_path} '''\n{contents}\n'''"

with open("full_code.txt", 'w') as output_file:
    output_file.write(output_string)
